politeiawww/cmd/piwww/testrun.go

```diff
@@ -446,15 +475,438 @@ func testUserRoutes(admin testUser, minPasswordLength int) error {
+	if err != nil {
+		return err
+	}
+
+	// Submit new proposal
+	pr, err := submitNewProposal(false)
+	if err != nil {
+		return err
+	}
 
+	// This is the proposal that we'll use for comment routes tests
+	token := pr.CensorshipRecord.Token
+	fmt.Printf("  Proposal submitted: %v\n", token)
+
+	// Login with admin and make the proposal public
+	fmt.Printf("  Login admin\n")
+	err = login(admin)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("  Set proposal status: public\n")
+	const proposalStatusPublic = "public"
+	err = proposalSetStatus(token, proposalStatusPublic, true)
+	if err != nil {
+		return err
+	}
+
+	// Log back in with user
+	fmt.Printf("  Login user\n")
+	err = login(*user)
+	if err != nil {
+		return err
+	}
+
+	// New comment - parent
+	fmt.Printf("  New comment: parent\n")
+	ncc := commentNewCmd{}
+	ncc.Args.Token = token
+	ncc.Args.Comment = "this is a comment"
+	ncc.Args.ParentID = "0"
+	err = ncc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	// New comment - reply
+	fmt.Printf("  New comment: reply\n")
+	ncc.Args.Token = token
+	ncc.Args.Comment = "this is a comment reply"
+	ncc.Args.ParentID = "1"
+	err = ncc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	// Validate comments
+	fmt.Printf("  Proposal details\n")
+	propReq := pi.ProposalRequest{
+		Token: token,
+	}
+	pdr, err := client.Proposals(pi.Proposals{
+		State:        pi.PropStateVetted,
+		Requests:     []pi.ProposalRequest{propReq},
+		IncludeFiles: false,
+	})
+	if err != nil {
+		return err
+	}
+	prop := pdr.Proposals[token]
+	if prop.Comments != 2 {
+		return fmt.Errorf("proposal num comments got %v, want 2",
+			prop.Comments)
+	}
+
+	fmt.Printf("  Proposal comments\n")
+	gcr, err := client.Comments(pi.Comments{
+		Token: token,
+		State: pi.PropStateVetted,
+	})
+	if err != nil {
+		return fmt.Errorf("Comments: %v", err)
+	}
+
+	if len(gcr.Comments) != 2 {
+		return fmt.Errorf("num comments got %v, want 2",
+			len(gcr.Comments))
+	}
+
+	for _, v := range gcr.Comments {
+		// We check the userID because userIDs are not part of
+		// the politeiad comment record. UserIDs are stored in
+		// in politeiawww and are added to the comments at the
+		// time of the request. This introduces the potential
+		// for errors.
+		if v.UserID != user.ID {
+			return fmt.Errorf("comment userID got %v, want %v",
+				v.UserID, user.ID)
+		}
+	}
+
+	// Login with admin to be able to vote on user's comments
+	fmt.Printf("  Login admin\n")
+	err = login(admin)
+	if err != nil {
+		return err
+	}
+
+	// Comment vote sequence
 	const (
 		// Comment actions
 		commentActionUpvote   = "upvote"
 		commentActionDownvote = "downvote"
 	)
+	cvc := commentVoteCmd{}
+	cvc.Args.Token = pr.CensorshipRecord.Token
+	cvc.Args.CommentID = "1"
+	cvc.Args.Vote = commentActionUpvote
+
+	fmt.Printf("  Comment vote: upvote\n")
+	err = cvc.Execute(nil)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("  Comment vote: upvote\n")
```

Comment:
@lukebp currently politeiad crashing at this level (second upvote from same user)